zkevm-circuits/src/evm_circuit/execution/coinbase.rs

```diff
@@ -0,0 +1,174 @@
+use crate::{
+    evm_circuit::{
+        execution::ExecutionGadget,
+        step::ExecutionState,
+        table::BlockContextFieldTag,
+        util::{
+            common_gadget::SameContextGadget,
+            constraint_builder::{
+                ConstraintBuilder, StepStateTransition, Transition::Delta,
+            },
+            Cell, Word,
+        },
+        witness::{Block, Call, ExecStep, Transaction},
+    },
+    util::Expr,
+};
+use bus_mapping::eth_types::ToLittleEndian;
+use halo2::{arithmetic::FieldExt, circuit::Region, plonk::Error};
+
+#[derive(Clone, Debug)]
+pub(crate) struct CoinbaseGadget<F> {
+    same_context: SameContextGadget<F>,
+    coinbase_address: Cell<F>,
```

Comment:
Need to use `RandomLinearCombination` as the look up in the stack needs to random linear combination of all bytes.

Reply:
yes, the random linear combination already do in block context's table_assignments() method of zkevm-circuits/src/evm_circuit/witness.rs